### PR TITLE
lib: Add new SRTM_percent color table

### PR DESCRIPTION
**Pull request by cmbarton**
This color table uses the color ramp from the srtm_plus color table for terrain to create colors for relative elevation (spread over the range of elevations in a raster map) rather than absolute elevation in meters. This applies srtm colors in a way similar to the way the elevation color table applies them. This color table is especially useful in creating nice looking elevation maps and shading relief maps.

If this seems useful, it could be backported to 8.4.x and 7.8

### CODE REVIEW

**Comment by neteler** [comment]
Would you mind to add a screenshot here? (simply pasting works)

**Comment by cmbarton** [comment]
> Would you mind to add a screenshot here? (simply pasting works)

Here are 2 from the new Flagstaff, AZ demo data set (NC does not have much relief). An SRTM DEM and a color-shaded relief map of the same area. 

flagstaff_dem
flagstaff_relief

**Comment by cmbarton** [comment]
Following up creating an image for Markus, I'd like to make a slight change in this. What is the best way to do it? Modify the PR or submit a new one?

**Comment by neteler** [comment]
Please modify this pull request.

**Comment by cmbarton** [comment]
> Please modify this pull request.

done. Added a new image too

**Comment by neteler** [comment]
For full integration of the new color table, it also needs to be added to https://github.com/OSGeo/grass/blob/main/lib/gis/colors.desc (otherwise it won't show up in `r.colors`, `v.colors`, etc.).

**Comment by cmbarton** [comment]
OK. Thanks. Just put a 2nd copy there?
____________________
Michael Barton

**Comment by neteler** [comment]
> OK. Thanks. Just put a 2nd copy there?

Essentially you edit the file here in your branch and commit/push it to this branch.

**Comment by echoix** [comment]
Since we normally squash-merge PRs, up until this PR is merged, you can add more commits, merge main into, do anything you'd like, and we can edit the final merge message once done

**Review by wenzeslaus — COMMENTED** [review]
The ds store modification to gitignore needs to be universal to all directories. That  might the thing which is better handled in a separate PR.

**Comment by nilason** [comment]
> The ds store modification to gitignore needs to be universal to all directories. That might the thing which is better handled in a separate PR.

Yes, that should go into a separate PR. Note that the pattern ".DS_Store" is enough.

It is also good to have this in the global git config:

```sh
echo .DS_Store >> ~/.gitignore
git config --global core.excludesfile ~/.gitignore
```

File: lib/gis/colors.desc [hunk 1]
```diff
@@ -48,6 +48,7 @@ sepia: yellowish-brown through to white
 slope: r.slope.aspect-type slope colors for raster values 0-90
 soilmoisture: soilmoisture color table (0.0-1.0)
 srtm: color palette for Shuttle Radar Topography Mission elevation
+srtm_pct: modified color palette for Shuttle Radar Topography Mission using relative elevation (0-100%) within a map instead of absolute elevation in meters
```
Comment (petrasovaa) [inline lib/gis/colors.desc:51]: Just to keep the description length consistent, it should be shorter, something like:
```suggestion
srtm_pct: relative color palette for Shuttle Radar Topography Mission
```
notice in r.colors man page it will add "[range: map values]" automatically

**Review by petrasovaa — COMMENTED** [review]
I am not fond of the name "srtm_pct", how about srtm_percent or srtm_relative?

**Comment by cmbarton** [comment]
1. I'm fine with shortening the description and will see about the suggested alternative. Any guidelines on how short/long? 
2. I originally had srtm_rel or srtm_relative but thought that was not clear. srtm_percent is fine. I was trying to keep it short. 
3. I don't want my .gitignore update to be in the PR. I did not intend for it to be there. So I am happy to take it out. How to do that? I'd like to set it for any future commits but don't understand how to set it for me and not have it go into grass/main. This is not a huge issue but suggestions would be welcome. 
4. Is 1 fork = 1 PR? Perhaps this is where I am confused. I was assuming that I could use my fork for >1 PR. Which is correct? 
5. What do you recommend I do at this point? 
- If there a way to delete the .gitignore part, is the new color table AND description already in the PR? 
- Or is the color.desc part still waiting to create a new PR? 
- It both parts (color table and description) is already in the PR, can I just rename the color table and modify the description locally, commit and push? 
- Do I have to do something else?

**Comment by echoix** [comment]
> 4. Is 1 fork = 1 PR? Perhaps this is where I am confused. I was assuming that I could use my fork for >1 PR. Which is correct? 
>

1 branch (of your fork) = 1 PR. You can continue changing the branch, and each commit pushed once the PR is opened shows up here. You can use your fork for creating multiple PRs, each in a branch.

**Comment by cmbarton** [comment]
> > 4. Is 1 fork = 1 PR? Perhaps this is where I am confused. I was assuming that I could use my fork for >1 PR. Which is correct?
> 
> 1 branch (of your fork) = 1 PR. You can continue changing the branch, and each commit pushed once the PR is opened shows up here. You can use your fork for creating multiple PRs, each in a branch.

Thanks much. This explains a lot. So I can make any updates, commit and push, and it will update the PR. Good to know.

**Comment by cmbarton** [comment]
I made the recommended changes in the PR. I am happy to remove the .gitignore commit if someone can tell me how. Can/should I just delete it in my fork/branch and commit/push?

**Comment by echoix** [comment]
> I made the recommended changes in the PR. I am happy to remove the .gitignore commit if someone can tell me how. Can/should I just delete it in my fork/branch and commit/push? 

That would do it. It's better a new commit (rather than rebase+force push) as there are already some review comments. After a force-push, it's hard to follow the comments that were made on commits that don't exist anymore in the PR.

**Comment by cmbarton** [comment]
> > I made the recommended changes in the PR. I am happy to remove the .gitignore commit if someone can tell me how. Can/should I just delete it in my fork/branch and commit/push?
> 
> That would do it. It's better a new commit (rather than rebase+force push) as there are already some review comments. After a force-push, it's hard to follow the comments that were made on commits that don't exist anymore in the PR.

Thanks. Done.

**Comment by wenzeslaus** [comment]
The goal is to remove the changes, not file file. Try `git checkout upstream/main -- path/to/file`. Upstream may be called differently depending on how you set up your local repo, check `git remote -v`.

**Comment by cmbarton** [comment]
> The goal is to remove the changes, not file file. Try `git checkout upstream/main -- path/to/file`. Upstream may be called differently depending on how you set up your local repo, check `git remote -v`.

I can always get this back locally. I just want this PR to go through

**Comment by petrasovaa** [comment]
Also, I just modified the PR title, see the instructions:
https://github.com/OSGeo/grass/blob/main/doc/development/github_guide.md#pr-title

**Review by echoix — APPROVED** [review]

**Review by petrasovaa — APPROVED** [review]
